waspc/data/Generator/templates/react-app/src/queries/index.js

```diff
@@ -1,18 +1,16 @@
 import { useQuery as rqUseQuery } from 'react-query'
 
-export const useQuery = (queryFn, queryFnArgs, config) => {
+export function useQuery(queryFn, queryFnArgs, config) {
   if (typeof queryFn !== 'function') {
-    throw new Error('useQuery requires queryFn to be a function.')
+    throw new TypeError('useQuery requires queryFn to be a function.')
   }
   if (!queryFn.queryCacheKey) {
-    throw new Error('queryFn needs to have queryCacheKey property defined.')
+    throw new TypeError('queryFn needs to have queryCacheKey property defined.')
   }
 
-  const rqResult = rqUseQuery({
+  return rqUseQuery({
     queryKey: [queryFn.queryCacheKey, queryFnArgs],
-    queryFn: (_key, args) => queryFn(args),
-    config
+    queryFn: () => queryFn(queryFnArgs),
+    ...config
```

Comment:
Reference: https://react-query.tanstack.com/guides/migrating-to-react-query-3#reactqueryconfigprovider-and-reactquerycacheprovider-have-both-been-replaced-by-queryclientprovider

Reply:
Should we also use collapsed config then? Or should we have it as separate config object? Maybe it is ok to have it separate in order to avoid conflict between react-query and our stuff. But likely that will never we a problem.

Reply:
We currently can't choose between collapsed/separate because our interface excepts positional arguments instead of objects. More details in the [other thread](https://github.com/wasp-lang/wasp/pull/554#discussion_r847449529).